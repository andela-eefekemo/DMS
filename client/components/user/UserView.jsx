import React from 'react';
import { Modal } from 'react-materialize';

const UserView = ({ id, firstName, lastName, email, deleteUser }) => {
  return (
    <div className="card blue-grey darken-1">
      <div className="card-content white-text">
        <span className="card-title">{firstName} {lastName}</span>
        <div>{email}</div>
      </div>
      <div className="card-action">
        <div className="row">
          <div className="col l12 m12 s12">
            <Modal
              trigger={
                <a
                  className="waves-effect btn button-design"
                  data-target="passwordModal">
                  Delete
                </a>}
              actions={
                <div>
                  <button className="btn waves-effect waves-light btn-flat modal-action modal-close" name={id} onClick={deleteUser}>
                    Delete
                </button>
                  <button className="btn waves-effect waves-light btn-flat modal-action modal-close left">Cancel</button>
                </div>}>
              <div >
                <h5 className="center">Are you sure you want to delete the user</h5>
              </div>
            </Modal>
          </div>
        </div>
      </div>
    </div>
  );
};

export default UserView;
