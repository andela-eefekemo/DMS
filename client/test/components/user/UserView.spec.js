/* global expect jest test */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import UserView from '../../../components/user/UserView';

jest.dontMock('../../../components/user/UserView');

describe('UserView Component', () => {
  test('should match the user view snapshot', () => {
    const deleteUser = jest.fn();
    const component = shallow(
      <UserView
        id={3}
        firstName={'Eguono'}
        lastName={'Efe'}
        email={'efe@gmail.com'}
        deleteUser={deleteUser} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
