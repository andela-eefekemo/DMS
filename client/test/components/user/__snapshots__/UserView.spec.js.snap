// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView Component should match the user view snapshot 1`] = `
<div
  className="card blue-grey darken-1"
>
  <div
    className="card-content white-text"
  >
    <span
      className="card-title"
    >
      Eguono
       
      Efe
    </span>
    <div>
      efe@gmail.com
    </div>
  </div>
  <div
    className="card-action"
  >
    <div
      className="row"
    >
      <div
        className="col l12 m12 s12"
      >
        <Modal
          actions={
            <div>
              <button
                className="btn btn-flat modal-action modal-close"
                name={3}
                onClick={[Function]}
              >
                Delete
              </button>
              <button
                className="btn btn-flat modal-action modal-close left"
              >
                Cancel
              </button>
            </div>
          }
          bottomSheet={false}
          fixedFooter={false}
          modalOptions={Object {}}
          trigger={
            <a
              className="waves-effect btn button-design"
              data-target="passwordModal"
            >
              Delete
            </a>
          }
        >
          <div>
            <h5
              className="center"
            >
              Are you sure you want to delete the user
            </h5>
          </div>
        </Modal>
      </div>
    </div>
  </div>
</div>
`;
